adam/ontology/ontology.py

```diff
@@ -61,6 +62,11 @@ class Ontology:
     action_to_description: ImmutableDict[OntologyNode, ActionDescription] = attrib(
         converter=_to_immutabledict, default=immutabledict(), kw_only=True
     )
+    _node_to_relations: ImmutableSetMultiDict[
```

Comment:
I would suggest just taking an `ImmutableSet` of relations; we can always index them later if we need to.